### ISSUE TITLE
trainerroad.py - RuntimeError: Failed to verify numerical key Weight

### ISSUE THREAD

**Issue by rattlesnailrick** [jaroslawhartman/withings-sync#83]
The script can properly read the current weight from withings and previous weight from trainerroad, but fails to update trainerroad.

```
2022-10-05 14:25:07,726 - withings - INFO - Refresh Access Token
2022-10-05 14:25:08,024 - root - INFO - Fetching measurements from 2022-10-05 00:00 to 2022-10-05 23:59
2022-10-05 14:25:08,236 - withings - INFO - Get Measurements
2022-10-05 14:25:08,443 - withings - INFO - Saving Last Sync
2022-10-05 14:25:08,444 - root - INFO - Trainerroad username set -- attempting to sync
2022-10-05 14:25:08,444 - root - INFO -  Last weight 86.1
2022-10-05 14:25:08,445 - root - INFO -  Measured 2022-10-05 06:41:57
2022-10-05 14:25:09,071 - withings_sync.trainerroad - INFO - Logged into TrainerRoad as "MYUSERNAME"
2022-10-05 14:25:10,240 - root - INFO - Current TrainerRoad weight: 86.5 kg
2022-10-05 14:25:10,244 - root - INFO - Updating TrainerRoad weight to 86.1 kg
2022-10-05 14:25:11,269 - withings_sync.trainerroad - INFO - Read profile values {'Ftp': '267', 'Weight': '86.5', 'Marketing': 'true', 'TimeZoneId': 'W. Europe Standard Time', 'IsPrivate': 'True', 'Units': 'kmh', 'IsVirtualPowerEnabled': 'False'}
2022-10-05 14:25:11,270 - withings_sync.trainerroad - INFO - New profile values {'Ftp': '267', 'Weight': '86.1', 'Marketing': 'true', 'TimeZoneId': 'W. Europe Standard Time', 'IsPrivate': 'True', 'Units': 'kmh', 'IsVirtualPowerEnabled': 'False'}
2022-10-05 14:25:13,532 - withings_sync.trainerroad - INFO - Read profile values (verification) {'Ftp': '267', 'Weight': '86.5', 'Marketing': 'true', 'TimeZoneId': 'W. Europe Standard Time', 'IsPrivate': 'True', 'Units': 'kmh', 'IsVirtualPowerEnabled': 'False'}
Traceback (most recent call last):
  File "/usr/local/bin/withings-sync", line 33, in <module>
    sys.exit(load_entry_point('withings-sync==3.4.2', 'console_scripts', 'withings-sync')())
  File "/usr/local/lib/python3.10/site-packages/withings_sync-3.4.2-py3.10.egg/withings_sync/sync.py", line 396, in main
    sync()
  File "/usr/local/lib/python3.10/site-packages/withings_sync-3.4.2-py3.10.egg/withings_sync/sync.py", line 365, in sync
    if sync_trainerroad(last_weight):
  File "/usr/local/lib/python3.10/site-packages/withings_sync-3.4.2-py3.10.egg/withings_sync/sync.py", line 147, in sync_trainerroad
    t_road.weight = round(last_weight, 1)
  File "/usr/local/lib/python3.10/site-packages/withings_sync-3.4.2-py3.10.egg/withings_sync/trainerroad.py", line 180, in weight
    self._write_profile({self._weight: value})
  File "/usr/local/lib/python3.10/site-packages/withings_sync-3.4.2-py3.10.egg/withings_sync/trainerroad.py", line 154, in _write_profile
    raise RuntimeError('Failed to verify numerical key {}'
RuntimeError: Failed to verify numerical key Weight
```

**Comment by darrencope** [comment]
Having the same issue, and was going to open an Issue but I forgot. Thank you @rattlesnailrick for doing so.

**Comment by darrencope** [comment]
Just upgraded to the latest version and still having this issue. Is there any update on this one? Thanks.

**Comment by longstone** [comment]
fixed in #113
